core/templates/directives/headroom.directive.ts

```diff
@@ -0,0 +1,50 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
```

Comment:
As per offline discussion with @srijanreddy98, we realized that using `ngx-headroom` would be much cleaner and simpler than having a custom directive. Can you please discuss this with @srijanreddy98 and update the PR as needed?

Reply:
@DubeySandeep after a closer look at ngx-headroom, we have realised that it has a lot unwanted animations and no option to not have it (e2e tests wold be expecting those animations). I think we should go with our custom solution.

Reply:
I think srijan answered this.

Reply:
@srijanreddy98 I'm trying it [on the playground](https://wicky.nillia.ms/headroom.js/playroom/) and found it is the same as what we have as of now. (and there are many options to configure it) am I missing anything? (not sure!)

Reply:
```
import { NgxHeadroomOption } from 'ngx-headroom';

let options: NgxHeadroomOption = {
  "offset": 112,
  "tolerance": 6,
  "classes": {
    "initial": "animated",
    "pinned": "slideDown",
    "unpinned": "slideUp"
  }
};
```

Can you try this^?

Reply:
I found that we are using the following offset & tolerance (But I think 200 & 5 is better, I'm fine with either) :
https://github.com/oppia/oppia/blob/c34fc9668a289de4bf7b29894de7f89017c050e3/core/templates/pages/exploration-editor-page/exploration-editor-page.mainpage.html#L42-L44

Reply:
Issues with @ctrl/ngx-headroom (Srijan originally suggested this one):

https://user-images.githubusercontent.com/24855641/120084326-e8792380-c0ec-11eb-951c-e467b535ebd8.mp4

Issues with akash-gupt/ngx-headroom (https://github.com/akash-gupt/ngx-headroom) (I assume you are suggesting this):
No longer maintained. 
The author worked on this library for two months.
The last commit on the repo is made on 26 Dec 2019.
He has neither worked on the library nor updated the dependencies from this date.

Reply:
I'm fine with either, @srijanreddy98 Can you confirm whether its fine to use custom directive here?

Reply:
@DubeySandeep after a closer look at ngx-headroom, we have realised that it has a lot unwanted animations and no option to not have it (e2e tests wold be expecting those animations). I think we should go with our custom solution.

_Originally posted by @srijanreddy98 in https://github.com/oppia/oppia/pull/12766#discussion_r635710029_